ga4gh/datamodel/genotype_phenotype.py

```diff
@@ -0,0 +1,467 @@
+                                                        location + ">)")
+                # print(location_query)
+                results = G2PDataset.g.query(location_query)
+                if (G2PDataset.record_test):
+                    self.test_data(results)
+                location_rows = [row.asdict() for row in results]
+                for row in location_rows:
+                    for k in row:
+                        row[k] = row[k].toPython()
+        self.record_more(rows)
+
+        annotation = self.flatten(rows)
+        location = self.flatten(location_rows)
+        annotation['location'] = location
+        return annotation
+
+    def flatten(self, dict):
+        """
+        Given a dict of dicts,
+        flatten it to a single dict using predicate as keys
+        For multiple occurrences of a predicate, create an array
+        Each value in the dict is an object {val:'x', label:'y'}
+        The value of 's' (subject) is copied to the 'id' property
+        """
+        a = {}
+        for row in dict:
+            obj = {'val': row['o']}
+            if 'label' in row:
+                obj['label'] = row['label']
+
+            if (row['p'] in a and
+                    a[row['p']].__class__.__name__ != "list"):
+                as_is = a[row['p']]
+                a[row['p']] = []
+                a[row['p']].append(as_is)
+
+            if (row['p'] in a):
+                a[row['p']].append(obj)
+            else:
+                a[row['p']] = obj
+
+            a['id'] = row['s']
+        return a
+
+    def to_ga4gh(self, annotation):
+        """
+        given an annotation dict, return a protocol.FeaturePhenotypeAssociation
+        """
+
+        fpa = None
+
+        # annotation keys
+        # hasSubject = 'http://purl.org/oban/association_has_subject'
+        # hasPredicate = 'http://purl.obolibrary.org/obo/RO_has_environment'
+        source = 'http://purl.org/dc/elements/1.1/source'
+        location = 'location'
+        _evidence = 'http://purl.obolibrary.org/obo/RO_0002558'
+        hasObject = 'http://purl.org/oban/association_has_object'
+        # location keys
+        GENO_0000408 = 'http://purl.obolibrary.org/obo/GENO_0000408'
+        # GENO_reference_amino_acid = 'http://purl.obolibrary.org/obo/'
+        # 'GENO_reference_amino_acid'
+        # _type = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#type'
+        # _location = 'http://biohackathon.org/resource/faldo#location'
+        # GENO_results_in_amino_acid_change = 'http://purl.obolibrary.org/obo/'
+        # 'GENO_results_in_amino_acid_change'
+        # RO_0002200 = 'http://purl.obolibrary.org/obo/RO_0002200'
+        # RO_0002205 = 'http://purl.obolibrary.org/obo/RO_0002205'
+        # label = 'http://www.w3.org/2000/01/rdf-schema#label'
+
+        location = annotation['location']
+        if(GENO_0000408 in location):
+            (_id, _ontologySource) = self.namespace_split(
```

Comment:
The brackets aren't needed here. Why the leading underscores? For things like ``id`` we follow the common Python convention of ``id_`` to avoid clobbering the global function.